frontend/src/Crossword.jsx

```diff
@@ -4,14 +4,30 @@ import { useState } from "react";
 
 import "./Crossword.css";
 import { useApi } from "./api";
+import { squares, createLetterArr, createCoordGrid } from "./squares";
+
+import Grid from "./Grid.jsx";
+
+const coordGrid = createCoordGrid();
 
 const ERASERSIZE = 12;
 const BRUSHSIZE = 4;
 
+const DRAW = 0;
+const WRITE = 1;
+
 const Crossword = (props) => {
   const [canvas, setCanvas] = useState(null);
   const [context, setContext] = useState(null);
-  const [readyState, onExternalDraw, sendEvent] = useApi(props.url);
+
+  const [readyState, onExternalDraw, onExternalWrite, sendEvent] = useApi(
+    props.url
+  );
+
+  const [mode, setMode] = useState(WRITE);
+
+  const [cursorRC, setCursorRC] = useState([0, 0]);
```

Comment:
`RC` to `position`?

Reply:
Ok. Had it like this to be explicit about the ordering (row, column vs. x, y) but I guess there are enough usages in the code to make it clear